homeassistant/components/nextcloud/update.py

```diff
@@ -0,0 +1,51 @@
+"""Update data from Nextcoud."""
+from __future__ import annotations
+
+from homeassistant.components.update import UpdateEntity, UpdateEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import NextcloudDataUpdateCoordinator
+from .entity import NextcloudEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Nextcloud update entity."""
+    coordinator: NextcloudDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.data.get("system_update_available") is None:
+        return
```

Comment:
Is this variable only filled when there is an update available? Or only when the system has update capabilities?

Reply:
It's either filled with `false` or the count of updates available when update capabilities are available. See https://github.com/nextcloud/serverinfo/blob/v28.0.0/lib/SystemStatistics.php#L95+L100

Reply:
But doesn't this then only add the entity when there is a system update available? Shouldn't we always add the entity?